opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/SpanDecorator.java

```diff
@@ -77,16 +90,32 @@ public void onResponse(HttpServletRequest httpServletRequest, HttpServletRespons
         }
 
         @Override
-        public void onError(HttpServletRequest httpServletRequest, Throwable exception, Span span) {
+        public void onError(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
+                            Throwable exception, Span span) {
             Tags.ERROR.set(span, Boolean.TRUE);
             span.log(logsForException(exception));
+
+            if (httpServletResponse.getStatus() == HttpServletResponse.SC_OK) {
+                // exception is thrown in filter chain, but status code is incorrect
+                Tags.HTTP_STATUS.set(span, 500);
+            }
+        }
+
+        @Override
+        public void onTimeout(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
+                              long timeout,
+                              Span span) {
+            span.setTag("timeout", Boolean.TRUE);
```

Comment:
I forgot to respond..

Yes I think it should be an error. I have added and error tag and log with message and timeout.